spec/datadog/tracing/contrib/http/circuit_breaker_spec.rb

```diff
@@ -74,45 +74,19 @@
           it { is_expected.to be false }
         end
       end
-    end
-
-    context 'integration' do
-      context 'given a request from' do
-        let(:request) { @request }
-
-        subject(:send_traces) do
-          # Capture the HTTP request directly from the transport,
-          # to make sure we have legitimate example.
-          expect(::Net::HTTP::Post).to receive(:new).and_wrap_original do |m, *args|
-            @request = m.call(*args)
-          end
-
-          # The request may produce an error (because the transport cannot connect)
-          # but ignore this... we just need the request, not a successful response.
-          allow(Datadog.logger).to receive(:error)
-
-          # Send a request, and make sure we captured it.
-          transport.send_traces(get_test_traces(1))
-
-          expect(@request).to be_a_kind_of(::Net::HTTP::Post)
-        end
 
-        context 'a Datadog Net::HTTP transport' do
-          before { expect(::Net::HTTP).to receive(:new) }
-
-          let(:transport) { Datadog::Transport::HTTP.default }
+      context 'with the DD-Internal-Untraced-Request header' do
+        context 'is present' do
+          let(:request) { ::Net::HTTP::Post.new('/some/path', headers) }
+          let(:headers) { { 'DD-Internal-Untraced-Request' => 'anything' } }
 
           it { is_expected.to be true }
         end
 
-        context 'a Datadog UDS transport' do
-          let(:transport) do
-            Datadog::Transport::HTTP.default do |t|
-              t.adapter :unix, '/tmp/ddagent/trace.sock'
-            end
-          end
+        context 'is missing' do
+          let(:request) { ::Net::HTTP::Post.new('/some/path') }
 
-          it { is_expected.to be true }
+          it { is_expected.to be false }
```

Comment:
This diff is confusing because of the test removed: this PR did not change this assertion from `true` to `false`. See side-by-side for more clarity.